src/libs/ReportUtils.ts

```diff
@@ -929,57 +784,53 @@ function getDefaultWorkspaceAvatar(workspaceName) {
         .replace(/[^0-9a-z]/gi, '')
         .toUpperCase();
 
-    return !alphaNumeric ? defaultWorkspaceAvatars.WorkspaceBuilding : defaultWorkspaceAvatars[`Workspace${alphaNumeric[0]}`];
+    const defaultWorkspaceAvatar = defaultWorkspaceAvatars[`Workspace${alphaNumeric[0]}`] as React.FC<SvgProps>;
+
+    return !alphaNumeric ? defaultWorkspaceAvatars.WorkspaceBuilding : defaultWorkspaceAvatar;
 }
 
-function getWorkspaceAvatar(report) {
-    const workspaceName = getPolicyName(report, allPolicies);
-    return lodashGet(allPolicies, [`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`, 'avatar']) || getDefaultWorkspaceAvatar(workspaceName);
+function getWorkspaceAvatar(report: OnyxEntry<Report>) {
+    const workspaceName = getPolicyName(report, false, allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`]);
+    return allPolicies?.[`policy${report?.policyID}`]?.avatar ?? getDefaultWorkspaceAvatar(workspaceName);
 }
 
 /**
  * Returns the appropriate icons for the given chat report using the stored personalDetails.
  * The Avatar sources can be URLs or Icon components according to the chat type.
- *
- * @param {Array} participants
- * @param {Object} personalDetails
- * @returns {Array<*>}
  */
-function getIconsForParticipants(participants, personalDetails) {
-    const participantDetails = [];
+function getIconsForParticipants(participants: number[], personalDetails: OnyxCollection<PersonalDetails>) {
+    const participantDetails: Array<[number, string, string | React.FC<SvgProps>, React.FC<SvgProps> | string]> = [];
```

Comment:
@kubabutkiewicz Why do you have both? `string | React.FC<SvgProps>, React.FC<SvgProps> | string `

Reply:
@VickyStash because of this
<img width="951" alt="image" src="https://github.com/Expensify/App/assets/25230417/0e8a01db-3255-4fe2-9a0f-dc34643bb669">
`avatarSource` and `fallbackIcon` are type of `string | React.FC<SvgProps>`